components/browser/menu2/src/main/java/mozilla/components/browser/menu2/adapter/icons/MenuIconAdapter.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu2.adapter.icons
+
+import android.view.LayoutInflater
+import androidx.annotation.VisibleForTesting
+import androidx.constraintlayout.widget.ConstraintLayout
+import mozilla.components.browser.menu2.Side
+import mozilla.components.browser.menu2.candidate.DrawableButtonMenuIcon
+import mozilla.components.browser.menu2.candidate.DrawableMenuIcon
+import mozilla.components.browser.menu2.candidate.MenuIcon
+import mozilla.components.browser.menu2.candidate.TextMenuIcon
+
+/**
+ * Helper class to manage a [ConstraintLayout] that can contain menu icon views.
+ * Different holder classes are used to swap out the child views.
+ */
+internal class MenuIconAdapter(
+    private val parent: ConstraintLayout,
+    private val inflater: LayoutInflater,
+    private val side: Side,
+    private val dismiss: () -> Unit
+) {
+
+    private var viewHolder: MenuIconViewHolder<*>? = null
```

Comment:
```suggestion
    private var viewHolder: MenuIconViewHolder<out MenuIcon>? = null
```

Same comment here I think.